test/BaseTestEntityWithSources.cpp

```diff
@@ -82,4 +82,30 @@ void BaseTestEntityWithSources::testSourceVectorSetter() {
     std::vector<Source> deleter;
     da.sources(deleter);
     CPPUNIT_ASSERT(da.sourceCount() == 0);
-}
\ No newline at end of file
+}
+
+void BaseTestEntityWithSources::testSourceOrder() {
+    std::vector<std::string> names = { "source_a", "source_b", "source_c", "source_d", "source_e" };
```

Comment:
I agree it would be a better test. For instance, I can imagine a situation where the order isn't tracked but the Sources are returned in alphabetical order by name (say, in a future FS backend that links by name).
Randomising is a good way of ensuring tracking works as expected.